addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -158,7 +160,7 @@ static func get_categories(blocks: Array[Block], extra_categories: Array[BlockCa
 	# Accessing a static Callable from a static function fails in 4.2.1.
 	# Use the fully qualified name.
 	# https://github.com/godotengine/godot/issues/86032
-	cats.sort_custom(CategoryFactory._category_cmp)
+	cats.sort_custom(_category_cmp)
```

Comment:
This invalidates the comment above it. If I read https://github.com/godotengine/godot/issues/86032, I think this will fail on 4.2.1 with an anonymous class. It would be worth testing in 4.2.1. Then I think either we keep the global `CategoryFactory` class or change the requirements to 4.2.2.

Reply:
Ohh, I'm blind today. Let's keep the global class for now, with a comment.